### PR TITLE
update codebase to support fastapi>=0.100.0 and pydantic>=2.0.0

### PR DESCRIPTION
**Pull request by johnybx**
Hi,
I updated codebase to support fastapi>=0.100.0 and pydantic>=2.0.0 ( this is breaking change ). I tried to keep same features and interface but when it comes to `with_prefix` function there were changes required. Currently with pydantic>=2.0.0 when you define attribute like in this example:
```python
        class MainFilter(BaseModel):
            name: str
            number_filter: Optional[NumberFilter] = FilterDepends(with_prefix("number_filter", NumberFilter))
```
you get error:
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for FilterWrapper
  address
    Input should be a valid dictionary or instance of NumberFilter [type=model_type, input_value=FilterWrapper(...), input_type=FilterDepends.<locals>.FilterWrapper]
  For further information visit https://errors.pydantic.dev/2.2/v/model_type
```
The only workaround I found (currently) is to update `with_prefix` to generate the wrapped model and annotation with `PlainValidator`.

TODO:
- test with all supported python versions
- update documentation
- this is breaking change - should probably update major version instead of minor ?

### CODE REVIEW

**Comment by johnybx** [comment]
I came up with different approach which does not introduce breaking changes to how  `with_prefix` is used. It would be nice if you could review the changes @arthurio. Thanks

File: fastapi_filter/base/filter.py [hunk 1]
```diff
@@ -212,18 +222,23 @@ def FilterDepends(Filter: Type[BaseFilterModel], *, by_alias: bool = False, use_
...
-        def sort(self, *args, **kwargs):
+        def __new__(cls, *args, **kwargs):
             try:
-                original_filter = Filter(**self.dict(by_alias=by_alias))
+                instance = GeneratedFilter(*args, **kwargs)
+                data = instance.model_dump(exclude_unset=True, exclude_defaults=True, by_alias=by_alias)
+                if original_filter := getattr(Filter.Constants, "original_filter", None):
+                    prefix = f"{Filter.Constants.prefix}__"
+                    stripped = {}
+                    k: str
```
Comment (arthurio) [inline fastapi_filter/base/filter.py:232]:
```suggestion
```

File: tests/test_mongoengine/test_order_by.py [hunk 2]
```diff
@@ -143,7 +147,8 @@ def test_restricted_order_by_success(User, UserFilterRestrictedOrderBy, order_by
     ],
 )
 @pytest.mark.asyncio
-async def test_api_order_by(test_client, users, order_by, assert_function):
+@pytest.mark.usefixtures("users")
```
Comment (arthurio) [inline tests/test_mongoengine/test_order_by.py:150]: TIL, that's nice 👍🏻

**Review by arthurio — CHANGES_REQUESTED** [review]
That look really great, thank you so much for all your work!
I just have a few nits but we can ship it after that 🚀

**Review by arthurio — COMMENTED** [review]
Just that one last question so that we avoid confusion for people hitting that edge case, otherwise looks like it's ready to go! 👍🏻

File: fastapi_filter/base/filter.py [hunk 3]
```diff
@@ -167,32 +171,38 @@ class MainFilter(BaseModel):
...
+        annotation = f.annotation
+
+        if get_origin(annotation) in UNION_TYPES:
+            annotation_args: list = list(get_args(f.annotation))
+            if type(None) in annotation_args:
+                annotation_args.remove(type(None))
+            if len(annotation_args) == 1:
+                annotation = annotation_args[0]
+            # Not sure what to do if there is more then 1 value 🤔
+            # Do we need to handle Optional[Annotated[...]] ?
```
[the remark this answers is not in the document]
Comment (arthurio) [reply fastapi_filter/base/filter.py:198]: @johnybx Do you have time to make that change or do you want me to do it?

**Comment by johnybx** [comment]
@arthurio sorry that it took so long but I just realized that `pending` label on comment does not mean that it is waiting for reviewer but rather that it is not submitted :facepalm:

**Review by arthurio — APPROVED** [review]
Let's do it! Thanks again for all the hard work, really appreciate it.

**Pull request merged**